conf/pipfrozen-bash_env_var9-19-14.txt
BeautifulSoup==3.2.1
Django==1.6.6
Jinja2==2.7.3
Markdown==2.4.1
MarkupSafe==0.23
MySQL-python==1.2.5
Pillow==2.5.3
Pygments==1.6
SQLAlchemy==0.9.7
South==1.0
Sphinx==1.2.3
ajaxuploader==0.3.8
amqp==1.4.6
anyjson==0.3.3
argparse==1.2.1
bcrypt==1.0.2
beautifulsoup4==4.3.2
billiard==3.3.0.18
bleach==1.4
celery==3.1.13
certifi==14.05.14
cffi==0.8.6
coverage==3.7.1
cx-Oracle==5.1.3
dingus==0.3.4
distribute==0.7.3
django-admin-bootstrapped==1.6.9
django-admin-tools==0.5.2
django-adminactions==0.6
django-ajax-selects==1.3.5
django-ajax-upload-widget==0.5.5
django-appconf==0.6
django-bootstrap-toolkit==2.15.0
django-bootstrap3==4.11.0
django-celery==3.1.10
django-compressor==1.4
django-crispy-forms==1.4.0
django-debug-toolbar==1.2.1
django-discover-runner==1.0
django-dtpanel-htmltidy==0.1.1
django-filter==0.7
django-floppyforms==1.2.0
django-redis==3.7.1
django-tables2==0.15.0
django-tastypie==0.11.1
djangorestframework==2.4.0
docutils==0.12
dropbox==2.1.0
flake8==2.2.3
googleapi==0.1.2
gunicorn==19.1.1
html5lib==1.0b3
kombu==3.0.21
mccabe==0.2.1
mechanize==0.2.5
newrelic==2.30.0.27
pep8==1.5.7
py==1.4.24
pycparser==2.10
pycurl==7.19.5
pyflakes==0.8.1
python-dateutil==2.2
python-memcached==1.53
python-mimeparse==0.1.4
pytidylib==0.2.3
pytz==2014.4
redis==2.10.3
requests==2.4.0
selenium==2.43.0
six==1.7.3
sorl-thumbnail==11.12.1b
sqlparse==0.1.11
tox==1.7.2
transifex-client==0.11.1.beta
unicodecsv==0.9.4
urllib3==1.9
virtualenv==1.11.6
wheel==0.24.0
wsgiref==0.1.2
xlrd==0.9.3
xlwt==0.7.5
     
     
BASH ENV VARS     
BFYPRDLOGIN=sqlplus -S prod_team_ro/9thfl00r@//borac101-vip.l3.bluefly.com:1521/bfyprd1
CDC_JOINED_DC=corpdc1.bluefly.corp
CDC_JOINED_DOMAIN=bluefly.corp
CDC_JOINED_SITE=default-first-site-name
CDC_JOINED_ZONE=DC=bluefly,DC=corp
CDC_LOCALHOST=prodimages.bluefly.corp
CDC_PREW2KHOST=prodimages
DATASRV=/mnt/Post_Ready/zProd_Server/imageServer7/data
DJANGO_COLORS=light
DSSPRDLOGIN=sqlplus -S jbragato/Blu3f!y@//192.168.30.66:1531/dssprd1
DYLD_LIBRARY_PATH=/usr/lib/oracle/11.2/client64/lib:/usr/lib/oracle/11.2/client64:/usr/lib/oracle/11.2/client64
EDITOR=/usr/bin/nano
GST_PLUGIN_PATH=/home/johnb/vm/ND14/gst-watermarkplugin
HOME=/home/johnb
IMGSELECTS=/mnt/Post_Ready/zProd_Server/imageServer7/images/images_jpg_photoselects
IPYTHONDIR=/home/johnb/.ipython
KRB5CCNAME=FILE:/tmp/krb5cc_cdc1103103127_W9s81q
LD_LIBRARY_PATH=/usr/lib/oracle/11.2/client64/lib
LESS=-r
LIBSRV=/mnt/Post_Ready/zProd_Server/imageServer7/lib
LIMBO=/mnt/Post_Ready/zProd_Server/imageServer7/tmp/limbo
LOGNAME=johnb
MAGICK_FOLDER=/mnt/Post_Ready/zProd_Server/imageServer7/var/magick
MAGICK_LOAD=/mnt/Post_Ready/zProd_Server/imageServer7/var/magick/magick_load
MAIL=/var/mail/johnb
MYSQL_TCP_PORT=3301
NPY_NO_DEPRECATED_API=NPY_1_7_API_VERSION
OLDPWD=/home/johnb/virtualenvs/DJDAM/src
ORACLE_HOME=/usr/lib/oracle/11.2/client64
ORACLE_SID=ORCL
PAGER=less
PATH=/home/johnb/virtualenvs/DJDAM/bin:/usr/bin:/home/johnb/virtualenvs/DJDAM/bin:/home/johnb/virtualenvs/DJDAM/src/djdam:/home/johnb/virtualenvs/DJDAM/src/djdam/settings:/home/johnb/virtualenvs/DJDAM/src:/usr/lib/oracle/11.2/client64/lib:/usr/lib/oracle/11.2/client64:/usr/lib/oracle/11.2/client64:/home/johnb/virtualenvs/DJDAM/src/djdam:/usr/lib/oracle/11.2/client64/bin:/usr/share/centrifydc/sbin:/usr/share/centrifydc/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/local/bin:/usr/local/batchRunScripts/bash:/usr/local/batchRunScripts/php:/usr/local/batchRunScripts/python:/usr/sbin:/usr/bin:/sbin:/bin
PHPRC=/etc/php5/apache2/php.ini
POSTCOMPLETE=/mnt/Post_Complete
POSTREADY=/mnt/Post_Ready
PRODSRV=/mnt/Post_Ready/zProd_Server/imageServer7
PS1=(DJDAM)${debian_chroot:+($debian_chroot)}\u@\h:\w\$ 
PWD=/home/johnb/virtualenvs/DJDAM/src/conf
PYTHONPATH=$PYTHONPATH:/home/johnb/virtualenvs/DJDAM/local/debug_toolbar_html
PYTHONSTARTUP=/usr/local/batchRunScripts/python/pythonstartup.py
PYTHONWARNINGS=default
SCRIPTS=/mnt/Post_Ready/zProd_Server/imageServer7/scripts
SHELL=/bin/bash
SHLVL=1
SQLPATH=/usr/lib/oracle/11.2/client64
SQLSCRIPTS=/mnt/Post_Ready/zProd_Server/imageServer7/scripts/sql
SSH_CLIENT=192.168.21.32 51696 22
SSH_CONNECTION=192.168.21.32 51696 192.168.20.59 22
SSH_TTY=/dev/pts/2
TERM=xterm
TNS_ADMIN=/usr/lib/oracle/11.2/client64/lib
TODAY=2014-09-18
USER=johnb
USER_PRINCIPAL_NAME=JohnB@BLUEFLY.CORP
VENVDJDAM=/home/johnb/virtualenvs/DJDAM
VIRTUAL_ENV=/home/johnb/virtualenvs/DJDAM
WEBWARE_ROOT=/home/johnb/Webware
WEBWARE_WORK=/home/johnb/Sites/WorkDir
WORKON_HOME=/home/johnb/virtualenvs
Webapps=/home/johnb/Sites/webapps-dev
XDG_SESSION_COOKIE=0c2fdd186af17e60a78d680f00000155-1408152842.712881-2079954464
ZIMAGES=/mnt/Post_Ready/zImages_1
_=/usr/bin/env
archFashion=/mnt/Post_Ready/Retouch_Fashion
archStill=/mnt/Post_Ready/Retouch_Still
drop_PHOTOSTUDIO=/mnt/Post_Ready/zProd_Server/imageServer7/tmp/imagedrop/photostudio/
drop_RETOUCHERS=/mnt/Post_Complete/Complete_to_Load/Drop_FinalFilesOnly
fileRepo=/mnt/Post_Ready/zProd_Server/imageServer7/lib/
pushFashion=/mnt/Post_Ready/eFashionPush
pushStill=/mnt/Post_Ready/aPhotoPush
tmpDeletePhoto=/mnt/Post_Ready/zProd_Server/tmpDeletePhoto
tmpPhotoSelects=/mnt/Post_Ready/zProd_Server/imageServer7/tmp/tmpPhotoSelects
view_events=SELECT * FROM `data_imports`.`v.event_upcoming_incomplete` ORDER BY `v.event_upcoming_incomplete`.`event_start_dt` ASC
view_push_brand=SELECT * FROM `data_imagepaths`.`viewcount_prdsnapshot_pushselects_brand` ORDER BY `brand` ASC
view_push_category=SELECT * FROM `data_imagepaths`.`viewcount_prdsnapshot_pushselects_category` ORDER BY `category` ASC
view_push_gender=SELECT * FROM `data_imagepaths`.`viewcount_prdsnapshot_pushselects_gender` ORDER BY `gender` ASC


djdam/settings.py
"""
Django settings for djdam project.

For more information on this file, see
https://docs.djangoproject.com/en/1.6/topics/settings/

For the full list of settings and their values, see
https://docs.djangoproject.com/en/1.6/ref/settings/
"""

# Build paths inside the project like this: os.path.join(BASE_DIR, ...)
import os
BASE_DIR = os.path.dirname(os.path.dirname(__file__))

MEDIA_ROOT = os.path.join(BASE_DIR, 'var','media')
if os.path.isdir(MEDIA_ROOT):
    pass
else:
    os.makedirs(MEDIA_ROOT, mode=0777)

STATIC_ROOT= os.path.join(BASE_DIR, 'var','static')
if os.path.isdir(STATIC_ROOT):
    pass
else:
    os.makedirs(STATIC_ROOT, mode=0777)


# Static files (CSS, JavaScript, Images)
# https://docs.djangoproject.com/en/1.6/howto/static-files/

MEDIA_URL = '/media/'
STATIC_URL = '/static/'


SITE_ID = 1

# Quick-start development settings - unsuitable for production
# See https://docs.djangoproject.com/en/1.6/howto/deployment/checklist/

# SECURITY WARNING: keep the secret key used in production secret!
SECRET_KEY = 'quu&b1+7x+h511z5u&yfu)%9-^px86rzjuahd$t&1)iqepxz#l'

# SECURITY WARNING: don't run with debug turned on in production!
DEBUG = True

TEMPLATE_DEBUG = True

# Hosts/domain names that are valid for this site; required if DEBUG is False
# See https://docs.djangoproject.com/en/1.5/ref/settings/#allowed-hosts
# All hosts allowed
ALLOWED_HOSTS = ['*']
# No hosts allowed
# ALLOWED_HOSTS = []

## Must be set for Debug toolbar
INTERNAL_IPs = ('127.0.0.1',)


# Application definition

INSTALLED_APPS = (# '',
    # 'admin_bootstrap',
    # # DjangoAdminBootstrapped
    'django_admin_bootstrapped',
    # Dj Admin Tools
    #'admin_tools',  #
    #'admin_tools.theming',  #
    #'admin_tools.dashboard',  #
    #'admin_tools.menu',
    #'adminactions',  #
    # Django contrib apps/admin
    'django.contrib.auth',
    'django.contrib.sites',
    'django.contrib.admin',
    #'django.contrib.admin.apps.SimpleAdminConfig',
    'django.contrib.contenttypes',
    'django.contrib.sessions',
    'django.contrib.messages',
    'django.contrib.admindocs',
    #'django.contrib.markup',
    'django.contrib.humanize',
    'django.contrib.syndication',
    'django.contrib.staticfiles',
    'django.contrib.redirects',
    # Third-party apps, patches, fixes
    'djcelery',  #
    'compressor',

    ## Debug toolbar
    # 'debug_toolbar.apps.DebugToolbarConfig',  ## Uncomment for >= Django 1.7
    'debug_toolbar',  ## Uncomment for <= Django 1.6
    'debug_toolbar_htmltidy',
    #'debug_toolbar_user_panel',#


    # bootstrap and other toolkits for widgets tables forms etc.
    'bootstrap3',
    'bootstrap_toolkit',
    'crispy_forms',
    'django_tables2',
    'django_filters',
    'floppyforms',

    # Database migrations -- South unnecessary
    # 'south',

    # Gunicorn and Servers
    'gunicorn',

    # Image/Thumbnail Rendering and Manipulation
    # 'sorl.thumbnail',
    # 'ajaxuploader',
    # 'ajax_upload',

    ## REST
    'rest_framework',
    'tastypie',

    # Application base, containing global templates and Home Page View
    'base',

    # Local apps, referenced via appname
    'searcher',
    'imgadjust',
    # 'uploader',
    # 'accounts',

    # Local digassets app, referenced via appname
    #'digassets',

    # Local ajaxsearch app, referenced via appname
    'ajaxsearch',
    'ajax_select',
    #'autocomplete',
)

ROOT_URLCONF = 'djdam.urls'

WSGI_APPLICATION = 'djdam.wsgi.application'


# Database
# https://docs.djangoproject.com/en/1.6/ref/settings/#databases
DATABASES = {
    'default': {
        'ENGINE': 'django.db.backends.mysql',
        'NAME': 'www_django',
        'USER': 'root',
        'PASSWORD': 'mysql',
        'HOST': '127.0.0.1',
        'PORT': '3301',
        #'OPTIONS': {
        #    'init_command': 'SET storage_engine=InnoDB',
        #    'charset' : 'utf8',
        #    'use_unicode' : True,
        #},
        #'TEST_CHARSET': 'utf8',
        #'TEST_COLLATION': 'utf8_general_ci',
    },
    # 'slave': {
    #     ...
    # },
}

# Uncomment this and set to all slave DBs in use on the site.
# SLAVE_DATABASES = ['slave']

################## EMAIL SETTINGS AND LOCAL ADMIN ######################
# Recipients of traceback emails and other notifications.
ADMINS = (
    ('johnb', 'john.bragato@gmail.com'),
)
MANAGERS = ADMINS

# Internationalization
# https://docs.djangoproject.com/en/1.6/topics/i18n/

LANGUAGE_CODE = 'en-us'

TIME_ZONE = 'UTC'

USE_I18N = True

USE_L10N = True

USE_TZ = True


MIDDLEWARE_CLASSES = (
    'django.contrib.sessions.middleware.SessionMiddleware',
    'django.middleware.common.CommonMiddleware',
    'django.middleware.csrf.CsrfViewMiddleware',
    'django.contrib.auth.middleware.AuthenticationMiddleware',
    'django.contrib.messages.middleware.MessageMiddleware',
    'django.middleware.clickjacking.XFrameOptionsMiddleware',
    'django.contrib.redirects.middleware.RedirectFallbackMiddleware',
    'searcher.utils.Tastypie_Default_Format.TastyJSONMiddleware',
    'debug_toolbar.middleware.DebugToolbarMiddleware',
)



TEMPLATE_DIRS = (
    os.path.join(BASE_DIR, 'templates'),
    os.path.join(BASE_DIR, 'templates/searcher'),
    os.path.join(BASE_DIR, 'templates/uploader'),
    os.path.join(BASE_DIR, 'templates/ajaxsearch'),
    os.path.join(BASE_DIR, 'templates/accounts'),
    os.path.join(BASE_DIR, 'templates/imgadjust'),
)


# List of callables that know how to import templates from various sources.
TEMPLATE_LOADERS = (
    'django.template.loaders.filesystem.Loader',
    'django.template.loaders.app_directories.Loader',
    'django.template.loaders.eggs.Loader',
)

TEMPLATE_CONTEXT_PROCESSORS = (
    'django.contrib.auth.context_processors.auth',
    'django.core.context_processors.debug',
    'django.core.context_processors.i18n',
    'django.core.context_processors.media',
    'django.core.context_processors.static',
    'django.core.context_processors.request',
    'django.contrib.messages.context_processors.messages',
)

EMAIL_BACKEND = 'django.core.mail.backends.console.EmailBackend'
DEFAULT_FROM_EMAIL = 'john.bragato@gmail.com'
# EMAIL_FILE_PATH =
EMAIL_HOST = 'smtp.gmail.com'
EMAIL_PORT = '993'
EMAIL_HOST_USER = 'john.bragato@gmail.com'
EMAIL_HOST_PASSWORD = ''



#####################################################################
#################  SORL.THUMBNAIL & REDIS SERVER SETUP  #############
#####################################################################
CACHES = {
    'default': {
        'BACKEND': 'redis_cache.cache.RedisCache',
        'LOCATION': '127.0.0.1:6379:1',
        'OPTIONS': {
            'CLIENT_CLASS': 'redis_cache.client.DefaultClient',
        }
    }
}


THUMBNAIL_FORMAT = 'PNG'
THUMBNAIL_KVSTORE = 'sorl.thumbnail.kvstores.redis_kvstore.KVStore'
THUMBNAIL_REDIS_HOST = '127.0.0.1' # default
THUMBNAIL_REDIS_PORT = 6379 # default
THUMBNAIL_DEBUG = True
THUMBNAIL_QUALITY = 60
THUMBNAIL_PROGRESSIVE = False

FILE_UPLOAD_TEMP_DIR = '/var/tmp/'
#FILE_UPLOAD_TEMP_DIR = '/tmp'
FILE_UPLOAD_MAX_MEMORY_SIZE = 157286400

#####################################################################
#####       Storage/Cache Services
#####################################################################
#AWS_ACCESS_KEY_ID = os.environ.get('AWS_ACCESS_KEY_ID')
#AWS_SECRET_ACCESS_KEY = os.environ.get('AWS_SECRET_ACCESS_KEY')
#AWS_STORAGE_BUCKET_NAME = '<YOUR BUCKET NAME>'

#STATICFILES_STORAGE = 'storages.backends.s3boto.S3BotoStorage'
#DEFAULT_FILE_STORAGE = 'storages.backends.s3boto.S3BotoStorage'


# Log settings
import logging
logging.basicConfig()
log = logging.getLogger(__name__)

LOG_LEVEL = logging.INFO
HAS_SYSLOG = True

SYSLOG_TAG = "http_app_djdam"  # Make this unique to your project.

# Remove this configuration variable to use your custom logging configuration
LOGGING_CONFIG = None
LOGGING = {
   'version': 1,
   'loggers': {
       'djdam': {
           'level': "DEBUG"
       }
   }
}

# Place bcrypt first in the list, so it will be the default password hashing
# mechanism
PASSWORD_HASHERS = (
    'django.contrib.auth.hashers.BCryptPasswordHasher',
    'django.contrib.auth.hashers.PBKDF2PasswordHasher',
    'django.contrib.auth.hashers.PBKDF2SHA1PasswordHasher',
    'django.contrib.auth.hashers.SHA1PasswordHasher',
    'django.contrib.auth.hashers.MD5PasswordHasher',
    'django.contrib.auth.hashers.CryptPasswordHasher',
)

# Sessions
#
# By default, be at least somewhat secure with our session cookies.
SESSION_COOKIE_HTTPONLY = True

# Set this to true if you are using https
SESSION_COOKIE_SECURE = False

SESSION_SERIALIZER = 'django.contrib.sessions.serializers.JSONSerializer'

# Normally you should not import ANYTHING from Django directly
# into your settings, but ImproperlyConfigured is an exception.
# from django.core.exceptions import ImproperlyConfigured
#
#
# def get_env_setting(setting):
#     """ Get the environment setting or return exception """
#     try:
#         return os.environ[setting]
#     except KeyError:
#         error_msg = "Set the %s env variable" % setting
#         raise ImproperlyConfigured(error_msg)


# Specify a custom user model to use
#AUTH_USER_MODEL = 'accounts.djdamUser'
# SECURITY WARNING: keep the secret key used in production secret!
# Hardcoded values can leak through source control.
# This is an example method of getting the value from an environment setting.
# Uncomment to use, and then make sure you set the SECRET_KEY environment variable.
# This is good to use in production, and on services that support it such as Heroku.
#SECRET_KEY = get_env_setting('SECRET_KEY')



#==============================================================================
#  Celery settings
#==============================================================================
# import djcelery
# celery.setup loader()
# BROKER_URL = "django://"
# Uncomment these to activate and customize Celery:
# CELERY_ALWAYS_EAGER = False  # required to activate celeryd
# BROKER_HOST = 'localhost'
# BROKER_PORT = 5672
# BROKER_USER = 'django'
# BROKER_PASSWORD = 'django'
# BROKER_VHOST = 'django'
# CELERY_RESULT_BACKEND = 'amqp'

## New Celery Implementation
# import djcelery
# djcelery.setup_loader()
#
# BROKER_URL = 'redis://localhost:6379/0'
# CELERY_RESULT_BACKEND = 'redis://localhost:6379/0'
# CELERY_ACCEPT_CONTENT = ['json']
# CELERY_TASK_SERIALIZER = 'json'
# CELERY_RESULT_SERIALIZER = 'json'



# Enable these options for memcached
CACHE_BACKEND= "memcached://127.0.0.1:11211/"
CACHE_MIDDLEWARE_ANONYMOUS_ONLY=True

# Set this to true if you use a proxy that sets X-Forwarded-Host
USE_X_FORWARDED_HOST = True

#==============================================================================
#  Project Email settings
#==============================================================================

SERVER_EMAIL = "johnb@prodimages.ny.bluefly.com"
DEFAULT_FROM_EMAIL = "johnb@prodimages.ny.bluefly.com"
SYSTEM_EMAIL_PREFIX = "[prodimages.ny.bluefly.com]"

#==============================================================================
# Miscellaneous project settings
#==============================================================================

# Global Upload File Perms
FILE_UPLOAD_PERMISSIONS = 0664
# FILE_UPLOAD_PERMISSIONS = 1777

#==============================================================================
# Third party app settings
#==============================================================================
# List of callables that know how to import templates from various sources.


def custom_show_toolbar(request):
    """ Only show the debug toolbar to users with the superuser flag. """
    return request.user.is_superuser

DEBUG_TOOLBAR_CONFIG = {
    #'INTERCEPT_REDIRECTS': False,
    'SHOW_TOOLBAR_CALLBACK': 'djdam.settings.custom_show_toolbar',
    #'HIDE_DJANGO_SQL': False,
    'INSERT_BEFORE': 'body',
    'SHOW_TEMPLATE_CONTEXT': True,
    'ENABLE_STACKTRACES': True,
}
#
DEBUG_TOOLBAR_PANELS = (
    #'debug_toolbar_user_panel.panels.UsersPanel',
    'debug_toolbar.panels.request.RequestPanel',
    'debug_toolbar.panels.headers.HeadersPanel',
    'debug_toolbar.panels.sql.SQLPanel',
    'debug_toolbar.panels.templates.TemplatesPanel',
    'debug_toolbar_htmltidy.panels.HTMLTidyDebugPanel',
    'debug_toolbar.panels.staticfiles.StaticFilesPanel',
    'debug_toolbar.panels.signals.SignalsPanel',
    'debug_toolbar.panels.redirects.RedirectsPanel',
    # 'debug_toolbar.panels.cache.CachePanel',
    'debug_toolbar.panels.profiling.ProfilingPanel',
    'debug_toolbar.panels.versions.VersionsPanel',
    'debug_toolbar.panels.timer.TimerPanel',
    'debug_toolbar.panels.settings.SettingsPanel',
    'debug_toolbar.panels.logging.LoggingPanel',
)

### New Relic Monitoring Services Env Var for Startup script
# NEW_RELIC_CONFIG_FILE='newrelic.ini'


#
## Customize Admin interface with Admin Tools
#ADMIN_TOOLS_MENU = 'menu.CustomMenu'
#ADMIN_TOOLS_INDEX_DASHBOARD = 'dashboard.CustomIndexDashboard'
## ADMIN_TOOLS_APP_INDEX_DASHBOARD = 'dashboard.CustomAppIndexDashboard'
ADMIN_TOOLS_THEMING_CSS = 'css/theming.css'

CRISPY_TEMPLATE_PACK = 'bootstrap3'

## REST FRAMEWORK SETTINGS
TASTYPIE_DEFAULT_FORMATS = ['json']
API_LIMIT_PER_PAGE = 20

REST_FRAMEWORK = {
    'DEFAULT_PERMISSION_CLASSES': ('rest_framework.permissions.IsAdminUser',),
    'PAGINATE_BY': 20
}


### AUTOCOMPLETE django-ajax-selects
# define the lookup channels in use on the site
# AJAX_LOOKUP_CHANNELS = {
#     #  simple: search Person.objects.filter(name__icontains=q)
#     'person'  : {'model': 'example.person', 'search_field': 'name'},
#     # define a custom lookup channel
#     'song'   : ('example.lookups', 'SongLookup')
# }
